src/main/java/io/jenkins/pluginhealth/scoring/probes/LastCommitDateProbe.java

```diff
@@ -0,0 +1,78 @@
+package io.jenkins.pluginhealth.scoring.probes;
+
+import java.io.File;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+
+import io.jenkins.pluginhealth.scoring.model.Plugin;
+import io.jenkins.pluginhealth.scoring.model.ProbeResult;
+import io.jenkins.pluginhealth.scoring.model.ResultStatus;
+
+import org.slf4j.Logger;
+import org.eclipse.jgit.revwalk.RevCommit;
+import org.slf4j.LoggerFactory;
+import org.springframework.core.annotation.Order;
+import org.springframework.stereotype.Component;
+import org.eclipse.jgit.api.Git;
+import org.eclipse.jgit.api.errors.GitAPIException;
+
+/**
+ * Using the analysis done by {@link SCMLinkValidationProbe#}, this probe determines the last commit date of the plugin's repository.
+ */
+@Component
+@Order(value = LastCommitDateProbe.ORDER)
+public class LastCommitDateProbe extends Probe {
+    private static final Logger LOGGER = LoggerFactory.getLogger(LastCommitDateProbe.class);
+
+    public static final int ORDER = SCMLinkValidationProbe.ORDER + 1;
+
+    @Override
+    public ProbeResult doApply(Plugin plugin) {
+        final String scmLinkValidationProbeKey = new SCMLinkValidationProbe(null, null).key();
+
+        try {
+            if (plugin.getDetails().get(scmLinkValidationProbeKey) == null) {
+                LOGGER.error("Couldn't run {} on {} because previous SCMLinkValidationProbe has null value in database", key(), plugin.getName());
+                return ProbeResult.error(key(), "SCM link has not been probed yet");
+            }
+
+            if (plugin.getDetails().get(scmLinkValidationProbeKey).status().equals(ResultStatus.SUCCESS)) {
+                File dir = new File(System.getProperty("java.io.tmpdir") + "/" + plugin.getName());
```

Comment:
what about `Files.createTempDir` (not sure about the name, to be checked)?

Reply:
I think I tried that and it did not delete the content of the directory and the directory itself as well.
So, I included a custom `deleteDirectory` method to recursively the contents of the given directory and also delete the given directory itself in the end.

Reply:
temp files are normally deleted when you restart your computer.